lib/launcher.ts

```diff
@@ -0,0 +1,283 @@
+/**
+ * The launcher is responsible for parsing the capabilities from the
+ * input configuration and launching test runners.
+ */
+import * as q from 'q';
+import {Config, ConfigParser} from './configParser';
+import {Logger} from './logger2';
+import {Runner} from './runner';
+import {TaskRunner} from './taskRunner';
+import {TaskScheduler} from './taskScheduler';
+import * as helper from './util';
+
+let logger = new Logger('launcher');
+let RUNNERS_FAILED_EXIT_CODE = 100;
+
+/**
+ * Keeps track of a list of task results. Provides method to add a new
+ * result, aggregate the results into a summary, count failures,
+ * and save results into a JSON file.
+ */
+class TaskResults {
+  results_: any[] = [];
```

Comment:
Maybe for a follow-up CL, but it might make sense to introduce a class type for results.